### PR TITLE
[Refinement] refinement related to otaproxy launching

### PR DESCRIPTION
**Pull request by Bodong-Yang**
## Description
This PR introduces exposed helper functions for external launching otaproxy, supports configuring `cache_dir` and `cache_db_file` location for standalone otaproxy CLI.

### Motivation
1. Previously the standalone otaproxy startup cli (\_\_main__.py) **doesn't support configure cache_dir and cache_db_file location**, make the launching and testing standalone otaproxy harder as we have to modify the `otaproxy/config.py` every-time. 
2. Previously a wrapper for launching otaproxy in subprocess is implemented `ota_client_stubs`. This wrapper does the job of constructing otaproxy components, configuring ota_cache and otaproxy uvicorn server and laucnhing it in subprocess. **This reduces `ota_client_stub` module's cohesion and introduce deep coupling with otaproxy package, which is not ideal.**

### NOTE
The helper functions introduced by this PR is not yet used in main branch, it is used in refactored `ota_client_stub.py` in #212 . 

## Check list

- [x] test file(s) that covers the change(s) is implemented.
- [x] local test is passed. 

## Changes
at `otaclient.ota_proxy` package:
1.  `__init__.py`: add `run_otaproxy` helper function for setting up and starting otaproxy server, add wrapper function `subprocess_start_otaproxy` helper function for starting otaproxy as subprocess in one call.
4. `__main__.py`: add `--cache-dir` and `--cache-db-file` options for CLI interface. 

## Behavior changes

Does this PR introduce behavior change(s)?

- [ ] Yes, internal behaivor (will not impact user experience).
- [ ] Yes, external behaivor (will impact user experience).
- [x] No.

## Breaking change

Does this PR introduce breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by obi-t4** [comment]
@Bodong-Yang 
Could you explain why this PR is necessary? Or we will reject this PR.

**Comment by Bodong-Yang** [comment]
@obi-t4 (I also add the following lines into the PR comments)
1. Previously the standalone otaproxy startup cli (\_\_main__.py) doesn't support configure `cache_dir` and `cache_db_file` location, make the launching and testing standalone otaproxy harder as we have to modify the `config.py` everytime. so in this PR these options are provided as cli params.
2. **As a standalone python package, otaproxy package should expose helper functions for the user to launch otaproxy without knowing the implementation details(abstraction).** Previously, `ota_client_stub` launch otaproxy by directly importing internal components like `App` and `OTACache` classes from otaproxy, construct them into server instance, and launch it by `ota_client_stub` itself, **this reduce the cohesion** of `ota_client_stub`(it should only focusing on otaclient service API handling), **increase the coupling level with otaproxy package**(it constructs otaproxy server instance by itself), which is not ideal. This PR introduces some helper functions that user can use to launch otaproxy, without the need to let user compose otaproxy server instance by itself.

**Comment by obi-t4** [comment]
@Bodong-Yang I could understand the motivation and I'm thinking of introducing PR template.
I think we need motivation field for refinement PR.
I'd like to introduce the paragraph on the motivation section like:
- current behavior
- the problem about current behavior
- how to improve/fix the problem by this change

I can understand your motivation description, but to understand it more easily when we read this PR back later.

File: otaclient/ota_proxy/__init__.py [hunk 1]
```diff
@@ -13,9 +13,127 @@
...
+    to the _subprocess_main function, and then execute the function in
+    a subprocess.
+    check _subprocess_main function for more details.
+    """
+
+    # run otaproxy in async loop in new subprocess
+    mp_ctx = multiprocessing.get_context("spawn")
+    otaproxy_subprocess = mp_ctx.Process(
+        target=partial(_subprocess_main, *args, **kwargs),
+        daemon=True,  # kill otaproxy if otaclient exists
```
Comment (obi-t4) [inline otaclient/ota_proxy/__init__.py:101]: This function doesn't care about parent process is otaclient or not.
```suggestion
        daemon=True,  # kill otaproxy if the parent process exists
```

**Comment by obi-t4** [comment]
@Bodong-Yang 
> this reduce the cohesion of ota_client_stub(it should only focusing on otaclient service API handling), increase the coupling level with otaproxy package(it constructs otaproxy server instance by itself), which is not ideal. 

But this PR doesn't fix this issue. Step-by-step PR is OK, so I hope the next PR fix this problem.

**Comment by Bodong-Yang** [comment]
> But this PR doesn't fix this issue. Step-by-step PR is OK, so I hope the next PR fix this problem.

@obi-t4 Ah this is PR is split from #212 , so this PR only provides these functions, and the actual use is in the refactored `ota_client_stub.py` in #212 .

**Comment by obi-t4** [comment]
@Bodong-Yang 
Is it possible to split `the actual use is in the refactored ota_client_stub.py` by another PR?

**Comment by Bodong-Yang** [comment]
> @Bodong-Yang Is it possible to split `the actual use is in the refactored ota_client_stub.py` by another PR?

@obi-t4 It is difficult as the refactored `ota_client_stub.py` in #212  use the helper function `subprocess_start_otaproxy` in the first place(which means it doesn't use the legacy code in the beginning).

**Comment by Bodong-Yang** [comment]
( Also as we are introducing total refactoring of `ota_client_stub` in #212 , it is a meaningless to do modification on main branch's `ota_client_stub.py` I think.

**Comment by obi-t4** [comment]
OK, you need to add the description that the use of `subprocess_start_otaproxy` is done in #211.

**Comment by obi-t4** [comment]
@Bodong-Yang 
Can you rewrite the motivation section as follows:
```
- current behavior
- the problem about current behavior
- how to improve/fix the problem by this change
```

**Comment by Bodong-Yang** [comment]
@obi-t4 The PR description is updated to align the refinement PR template, thank you!

**Review by obi-t4 — APPROVED** [review]
LGTM :+1:

**Review by obi-t4 — APPROVED** [review]
LGTM :+1:

**Pull request merged**